koschei/backend/services/resolver.py

```diff
@@ -369,7 +369,7 @@ def persist_resolution_output(self, chunk):
         self.db.commit_no_expire()
 
         # emit fedmsg (if enabled)
-        if state_changes:
+        if state_changes and not repo_fixed:
```

Comment:
When repo is broken, packages are not resolved at all and thus retain their previous resolution state. The second bullet point is false, packages are not touched at all, only the collection is set as unresolved, and scheduler skips the packages based on the collection attribute. There is nothing to suppress.